diff --git a/CMakeLists.txt b/CMakeLists.txt
index 667b935..1c4d169 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,7 +6,7 @@ set(KF_DEP_VERSION "6.8.0") # handled by release scripts
 project(KQuickCharts VERSION ${KF_VERSION})
 
 include(FeatureSummary)
-find_package(ECM 6.8.0 NO_MODULE)
+find_package(ECM 6.7.0 NO_MODULE)
 set_package_properties(ECM PROPERTIES TYPE REQUIRED DESCRIPTION "Extra CMake Modules." URL "https://commits.kde.org/extra-cmake-modules")
 feature_summary(WHAT REQUIRED_PACKAGES_NOT_FOUND FATAL_ON_MISSING_REQUIRED_PACKAGES)
 
