diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7f1927d..f72a683 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -68,6 +68,9 @@ install(FILES
     COMPONENT Devel
 )
 
+install(EXPORT KF6QuickChartsTargets DESTINATION "${KDE_INSTALL_CMAKEPACKAGEDIR}/KF6QuickCharts"
+        FILE KF6QuickChartsTargets.cmake NAMESPACE KF6:: COMPONENT Devel)
+
 include(ECMFeatureSummary)
 ecm_feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES)
 
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index de8c77c..e22e83d 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -119,6 +119,7 @@ qt6_add_shaders(QuickCharts "shaders"
         piechart.vert.qsb
         piechart.frag.qsb
     ${_extra_args}
+    OUTPUT_TARGETS _out_targets
 )

 ecm_generate_export_header(QuickCharts
@@ -130,8 +131,8 @@ ecm_generate_export_header(QuickCharts
     EXCLUDE_DEPRECATED_BEFORE_AND_AT ${EXCLUDE_DEPRECATED_BEFORE_AND_AT}
 )

-ecm_finalize_qml_module(QuickCharts DESTINATION ${KDE_INSTALL_QMLDIR})
+ecm_finalize_qml_module(QuickCharts DESTINATION ${KDE_INSTALL_QMLDIR} EXPORT KF6QuickChartsTargets)

 ecm_qt_install_logging_categories(EXPORT KQuickCharts DESTINATION ${KDE_INSTALL_LOGGINGCATEGORIESDIR})

-install(TARGETS QuickCharts ${KDE_INSTALL_DEFAULT_ARGUMENTS})
+install(TARGETS QuickCharts ${_out_targets} EXPORT KF6QuickChartsTargets ${KF_INSTALL_TARGETS_DEFAULT_ARGS})
